crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -382,24 +307,27 @@ impl CheckpointExecutorEventLoop {
                 .unwrap_or(0),
         );
 
-        let task_limit = self.config.checkpoint_execution_max_concurrency;
-
         match next_to_exec.cmp(&latest_synced_checkpoint.sequence_number()) {
             // fully caught up case
-            Ordering::Greater => return Ok(()),
+            Ordering::Greater => {}
```

Comment:
Now that you have moved the concurrency check into the function, I believe the code can be a lot simpler if you just use a for loop for the entire thing, and then if you really want to optimize for the last db read, inside the loop you could do a comparison between i and latest_synced_checkpoint.sequence_number()

Reply:
Fixed